src/libraries/System.Private.CoreLib/src/System/DateTime.cs

```diff
@@ -1170,7 +1171,8 @@ public static int DaysInMonth(int year, int month)
         {
             if (month < 1 || month > 12) ThrowHelper.ThrowArgumentOutOfRange_Month(month);
             // IsLeapYear checks the year argument
-            return (IsLeapYear(year) ? DaysInMonth366 : DaysInMonth365)[month - 1];
```

Comment:
> return (IsLeapYear(year) ? DaysInMonth366 : DaysInMonth365)[month - 1]; [](http://example.com/codeflow?start=11&length=72)

I thought the jit should be optimizing the span indexing. Maybe something we can try instead of doing subtract 1 every time, we can have the span contents starts with 0 and avoid the subtraction operation and measure. 

Also, will be good to do the measurements on the real code and try to measure DateTime.DaysInMonth with and without the changes instead of writing a separate code mimic the library's code.

Reply:
CC @stephentoub

Reply:
The JIT can optimize and constant fold accesses to RVA static data (like the `DaysInMonth365`/`DaysInMonth366` data currently is).

Altogether, this new code seems a lot less readable for what is barely 1/5th of a nanosecond faster (basically 1-2 instruction cycles).

This is unlikely to make any difference in any real world code and I personally don't think the additional complexity is worth it here.

Reply:
I agree with @tannergooding here, if the perf gain is not going to make real difference in the public scenario, it is not worth the complexity.